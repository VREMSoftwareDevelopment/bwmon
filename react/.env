VITE_VERSION=$npm_package_version
VITE_NAME=$npm_package_name
VITE_DESCRIPTION=$npm_package_description
