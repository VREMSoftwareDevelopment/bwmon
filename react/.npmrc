save=true
puppeteer_skip_download=true
PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true