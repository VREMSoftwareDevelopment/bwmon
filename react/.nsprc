{
    "1070415": {
        "active": true,
        "expiry": "2022-12-31",
        "notes": "Ignored - waiting for update"
    }
}
