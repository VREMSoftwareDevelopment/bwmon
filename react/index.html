<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8" />
        <link rel="icon" href="/favicon.ico" />
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap" />
        <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons" />
        <meta name="viewport" content="minimum-scale=1, initial-scale=1, width=device-width" />
        <meta name="theme-color" content="#000000" />
        <meta name="Bandwidth Monitor" content="Bandwidth MonitorApplication" />
        <link rel="manifest" href="/manifest.json" />
        <base href="/" />
        <title>Bandwidth Monitor</title>
    </head>
    <body>
        <noscript>You need to enable JavaScript to run this app.</noscript>
        <div id="root"></div>
        <script type="module" src="/src/index.jsx"></script>
    </body>
</html>
