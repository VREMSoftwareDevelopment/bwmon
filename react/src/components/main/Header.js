/*
 *      Copyright (C) 2010 - 2020 VREM Software Development <VREMSoftwareDevelopment@gmail.com>
 *
 *      Licensed under the Apache License, Version 2.0 (the "License");
 *      you may not use this file except in compliance with the License.
 *      You may obtain a copy of the License at
 *
 *           http: //www.apache.org/licenses/LICENSE-2.0
 *
 *      Unless required by applicable law or agreed to in writing, software
 *      distributed under the License is distributed on an "AS IS" BASIS,
 *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *      See the License for the specific language governing permissions and
 *      limitations under the License.
 *
 * Bandwidth Monitor
 */

import React from 'react';
import { makeStyles } from '@mui/styles';
import { AppBar, Toolbar, Typography } from '@mui/material';

const useStyles = makeStyles({
    title: {
        flexGrow: 1,
    },
});

const Header = ({ name, version }) => {
    const classes = useStyles();

    return (
        <AppBar position="static">
            <Toolbar variant="dense">
                <Typography id="app-title" variant="h6" className={classes.title}>
                    {name}
                </Typography>
                <Typography id="app-version" className={classes.text}>
                    {version}
                </Typography>
            </Toolbar>
        </AppBar>
    );
};

export default Header;
