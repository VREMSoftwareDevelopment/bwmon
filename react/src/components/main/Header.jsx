/*
 *      Copyright (C) 2010 - 2024 VREM Software Development <VREMSoftwareDevelopment@gmail.com>
 *
 *      Licensed under the Apache License, Version 2.0 (the "License");
 *      you may not use this file except in compliance with the License.
 *      You may obtain a copy of the License at
 *
 *           http: //www.apache.org/licenses/LICENSE-2.0
 *
 *      Unless required by applicable law or agreed to in writing, software
 *      distributed under the License is distributed on an "AS IS" BASIS,
 *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *      See the License for the specific language governing permissions and
 *      limitations under the License.
 *
 * Bandwidth Monitor
 */

import React from 'react';
import PropTypes from 'prop-types';
import { AppBar, Toolbar, Typography } from '@mui/material';
import { headerStyles } from '../../utils/StylesUtils';

const Header = ({ name, version }) => {
    const classes = headerStyles();

    return (
        <AppBar position="static">
            <Toolbar variant="dense">
                <Typography data-testid="app-title" id="app-title" variant="h6" className={classes.title}>
                    {name}
                </Typography>
                <Typography data-testid="app-version" id="app-version" className={classes.text}>
                    {version}
                </Typography>
            </Toolbar>
        </AppBar>
    );
};

Header.propTypes = {
    name: PropTypes.string.isRequired,
    version: PropTypes.string.isRequired,
};

export default Header;
