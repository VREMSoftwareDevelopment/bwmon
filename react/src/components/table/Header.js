/*
 *      Copyright (C) 2010 - 2024 VREM Software Development <VREMSoftwareDevelopment@gmail.com>
 *
 *      Licensed under the Apache License, Version 2.0 (the "License");
 *      you may not use this file except in compliance with the License.
 *      You may obtain a copy of the License at
 *
 *           http: //www.apache.org/licenses/LICENSE-2.0
 *
 *      Unless required by applicable law or agreed to in writing, software
 *      distributed under the License is distributed on an "AS IS" BASIS,
 *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *      See the License for the specific language governing permissions and
 *      limitations under the License.
 *
 * Bandwidth Monitor
 */

import React from 'react';
import { TableCell, TableHead, TableRow, TableSortLabel } from '@mui/material';

const SortableCell = ({ prefix, cellInfo, sortHandler, ascending, orderBy }) => {
    const direction = ascending ? 'asc' : 'desc';
    const sortDirection = (orderBy, id) => (orderBy === id ? direction : false);
    const defaultDirection = (orderBy, id) => (orderBy === id ? direction : 'asc');

    return (
        <TableCell
            data-testid={prefix + '-' + cellInfo.id}
            id={prefix + '-' + cellInfo.id}
            align={cellInfo.align}
            sortDirection={sortDirection(orderBy, cellInfo.id)}
        >
            <TableSortLabel
                active={orderBy === cellInfo.id}
                direction={defaultDirection(orderBy, cellInfo.id)}
                onClick={sortHandler(cellInfo.id)}
            >
                {cellInfo.label}
            </TableSortLabel>
        </TableCell>
    );
};

const Cell = ({ cellInfo }) => <TableCell align={cellInfo.align}>{cellInfo.label}</TableCell>;

const Header = ({ prefix, cellInfos, onRequestSort, ascending, orderBy }) => {
    const sortHandler = (property) => (event) => {
        onRequestSort(event, property);
    };

    return (
        <TableHead>
            <TableRow key="header" data-testid={prefix + '-header'} id={prefix + '-header'}>
                {cellInfos.map((cellInfo) => {
                    return cellInfo.sortable ? (
                        <SortableCell
                            key={cellInfo.id}
                            prefix={prefix}
                            cellInfo={cellInfo}
                            sortHandler={sortHandler}
                            ascending={ascending}
                            orderBy={orderBy}
                        ></SortableCell>
                    ) : (
                        <Cell key={cellInfo.id} cellInfo={cellInfo} />
                    );
                })}
            </TableRow>
        </TableHead>
    );
};

export default Header;
