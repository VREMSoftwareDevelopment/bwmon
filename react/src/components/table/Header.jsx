/*
 *      Copyright (C) 2010 - 2024 VREM Software Development <VREMSoftwareDevelopment@gmail.com>
 *
 *      Licensed under the Apache License, Version 2.0 (the "License");
 *      you may not use this file except in compliance with the License.
 *      You may obtain a copy of the License at
 *
 *           http: //www.apache.org/licenses/LICENSE-2.0
 *
 *      Unless required by applicable law or agreed to in writing, software
 *      distributed under the License is distributed on an "AS IS" BASIS,
 *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *      See the License for the specific language governing permissions and
 *      limitations under the License.
 *
 * Bandwidth Monitor
 */

import React from 'react';
import PropTypes from 'prop-types';
import { TableCell, TableHead, TableRow, TableSortLabel } from '@mui/material';

const ASCENDING = 'asc';
const DESCENDING = 'desc';

const getDirection = (ascending) => (ascending ? ASCENDING : DESCENDING);
const sortDirection = (orderBy, id, ascending) => (orderBy === id ? getDirection(ascending) : false);
const sortDefaultDirection = (orderBy, id, ascending) => getDirection(ascending || orderBy !== id);
const isActive = (orderBy, id) => orderBy === id;

const SortableCell = ({ prefix, cellInfo, sortHandler, ascending, orderBy }) => (
    <TableCell
        data-testid={prefix + '-' + cellInfo.id}
        id={prefix + '-' + cellInfo.id}
        align={cellInfo.align}
        sortDirection={sortDirection(orderBy, cellInfo.id, ascending)}
    >
        <TableSortLabel
            active={isActive(orderBy, cellInfo.id)}
            direction={sortDefaultDirection(orderBy, cellInfo.id, ascending)}
            onClick={sortHandler(cellInfo.id)}
        >
            {cellInfo.label}
        </TableSortLabel>
    </TableCell>
);

// Stryker disable all
SortableCell.propTypes = {
    prefix: PropTypes.string.isRequired,
    cellInfo: PropTypes.shape({
        id: PropTypes.string.isRequired,
        align: PropTypes.string,
        label: PropTypes.string.isRequired,
        sortable: PropTypes.bool,
    }).isRequired,
    sortHandler: PropTypes.func.isRequired,
    ascending: PropTypes.bool.isRequired,
    orderBy: PropTypes.string.isRequired,
};
// Stryker restore all

const Cell = ({ cellInfo }) => <TableCell align={cellInfo.align}>{cellInfo.label}</TableCell>;

// Stryker disable all
Cell.propTypes = {
    cellInfo: PropTypes.shape({
        id: PropTypes.string.isRequired,
        align: PropTypes.string,
        label: PropTypes.string.isRequired,
        sortable: PropTypes.bool,
    }).isRequired,
};
// Stryker restore all

const Header = ({ prefix, cellInfos, onRequestSort, ascending, orderBy }) => {
    const sortHandler = (property) => (event) => {
        onRequestSort(event, property);
    };

    return (
        <TableHead>
            <TableRow key="header" data-testid={prefix + '-header'} id={prefix + '-header'}>
                {cellInfos.map((cellInfo) => {
                    return cellInfo.sortable ? (
                        <SortableCell
                            key={cellInfo.id}
                            prefix={prefix}
                            cellInfo={cellInfo}
                            sortHandler={sortHandler}
                            ascending={ascending}
                            orderBy={orderBy}
                        ></SortableCell>
                    ) : (
                        <Cell key={cellInfo.id} cellInfo={cellInfo} />
                    );
                })}
            </TableRow>
        </TableHead>
    );
};

// Stryker disable all
Header.propTypes = {
    prefix: PropTypes.string.isRequired,
    cellInfos: PropTypes.arrayOf(
        PropTypes.shape({
            id: PropTypes.string.isRequired,
            align: PropTypes.string,
            label: PropTypes.string.isRequired,
            sortable: PropTypes.bool,
        })
    ).isRequired,
    onRequestSort: PropTypes.func.isRequired,
    ascending: PropTypes.bool.isRequired,
    orderBy: PropTypes.string.isRequired,
};
// Stryker restore all

export { Header, Cell, SortableCell, getDirection, sortDirection, sortDefaultDirection, isActive };

export default Header;
