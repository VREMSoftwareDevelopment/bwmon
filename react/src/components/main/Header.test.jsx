/*
 *      Copyright (C) 2010 - 2025 VREM Software Development <VREMSoftwareDevelopment@gmail.com>
 *
 *      Licensed under the Apache License, Version 2.0 (the "License");
 *      you may not use this file except in compliance with the License.
 *      You may obtain a copy of the License at
 *
 *           http: //www.apache.org/licenses/LICENSE-2.0
 *
 *      Unless required by applicable law or agreed to in writing, software
 *      distributed under the License is distributed on an "AS IS" BASIS,
 *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *      See the License for the specific language governing permissions and
 *      limitations under the License.
 *
 * Bandwidth Monitor
 */

import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ThemeProvider, createTheme } from '@mui/material/styles';
import Header from './Header';

describe('Header', () => {
    const theme = createTheme();

    const renderComponent = (props) =>
        render(
            <ThemeProvider theme={theme}>
                <Header {...props} />
            </ThemeProvider>
        );

    it('renders application name and version', () => {
        const name = 'Name';
        const version = '3.1.1';
        renderComponent({ name: name, version: version });
        expect(screen.getByTestId('app-title')).toHaveTextContent(name);
        expect(screen.getByTestId('app-version')).toHaveTextContent(version);
    });
});
