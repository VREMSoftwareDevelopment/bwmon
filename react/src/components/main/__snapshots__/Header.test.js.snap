// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-5iifl-MuiPaper-root-MuiAppBar-root"
  style={
    Object {
      "--Paper-shadow": "0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12)",
    }
  }
>
  <div
    className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-dense css-203dff-MuiToolbar-root"
  >
    <h6
      className="MuiTypography-root MuiTypography-h6 makeStyles-title-1 css-8zgow4-MuiTypography-root"
      data-testid="app-title"
      id="app-title"
      style={Object {}}
    >
      Name
    </h6>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1edfpdg-MuiTypography-root"
      data-testid="app-version"
      id="app-version"
      style={Object {}}
    >
      3.1.1
    </p>
  </div>
</header>
`;
